stress/src/async_throughput.rs

```diff
@@ -0,0 +1,257 @@
+use futures::stream::{FuturesUnordered, StreamExt};
+use num_format::{Locale, ToFormattedString};
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+// # AsyncThroughputTest
+//
+// Measures the throughput and error rate of a user-provided asynchronous function under configurable load.
+//
+// ## What this measures
+//
+// - Number of successful and failed operations performed by an async function.
+// - Throughput (operations per second), reported live and as a final average.
+// - Success rate and error count during the test.
+//
+// ## How it works
+//
+// - Launches a configurable number of async worker tasks, each maintaining a set number of concurrent operations.
+// - As each operation completes, a new one is started to maintain steady concurrency.
+// - Successes and errors are tracked using atomic counters.
+// - A separate reporter task prints throughput and errors every second.
+// - The test can be stopped gracefully with Ctrl+C, allowing in-flight operations to finish.
+//
+// ## Typical use-cases
+//
+// - Load testing and benchmarking async functions, network clients, or service endpoints.
+// - Observing how throughput and error rates change under varying concurrency settings.
+//
+pub struct AsyncThroughputTest {
+    stats: Arc<SimpleStats>,
+    stop: Arc<AtomicBool>,
+}
+
+struct SimpleStats {
+    success_count: Arc<AtomicU64>,
+    error_count: Arc<AtomicU64>,
+}
+
+impl SimpleStats {
+    fn new() -> Self {
+        Self {
+            success_count: Arc::new(AtomicU64::new(0)),
+            error_count: Arc::new(AtomicU64::new(0)),
+        }
+    }
+}
+
+impl AsyncThroughputTest {
+    pub fn new() -> Self {
+        Self {
+            stats: Arc::new(SimpleStats::new()),
+            stop: Arc::new(AtomicBool::new(false)),
+        }
+    }
+
+    /// Run the throughput test with the given async function
+    ///
+    /// # Arguments
+    /// * `name` - Test name for display
+    /// * `workers` - Number of worker tasks
```

Comment:
This worker approach seems to be adding unnecessary overhead. The total concurrency (number of operations running simultaneously) is the only thing that matters for stress testing. The number of "worker" threads working on these operations should be decided by the tokio runtime flavor:
- One for `current_thread` runtime
- Multiple for `multi_thread` runtime

Reply:
Yes.

Reply:
The code is updated accordingly.